src/crimson/net/SocketConnection.h

```diff
@@ -27,13 +28,17 @@ class AuthSessionHandler;
 
 namespace ceph::net {
 
+using seastop = seastar::stop_iteration;
```

Comment:
done.

Reply:
I assume "search-n-replace" means refactoring inside `repeat_connect()` and `repeat_handle_connect()` to return `stop_t`.

Just tried to split but it is related to move `state = state_t::open` to `execute_open()`, which is part of "encapsulate protocol executions with states".

Reply:
no, i meant the change replacing `seastar::stop_iteration` with `stop_t`. in c046df3a30180ccf4201ec1e0c9b7af985ec9bbb , you have a handful changes:

1. replace `seastar::stop_iteration` with `stop_t`
1. return a `future<stop_t>` instead of `stop_t`. which i don't think it makes sense.
1. use messenger's `pending_dispatch` for guarding, presumably, dangling `dispatcher`?
1. throw an exception instead of calling `fault()` in `handle_connect_reply()`
1. rename `dispatch()` to `execute_open()`. does this new name imply a state ?

i think you might want to be more specific in each commit to help reviewers to review your changes.

Reply:
> encapsulate protocol executions with states.

how are the states encapsulated? are they different continuations or classes or states after this change?

Reply:
> 2. return a future<stop_t> instead of stop_t. which i don't think it makes sense.

Yeah, will fix that.

> 3. use messenger's pending_dispatch for guarding, presumably, dangling dispatcher?

Yep, but they are not "changes" I think... The guarding logics are kept intact in the commit, and were moved along with dispatching logics from `SocketMessenger` into `SocketConnection` in order to accomplish the goal of the commit c046df3. 

> 4. throw an exception instead of calling fault() in handle_connect_reply()

It would be **wrong** logic if we don't throw exception to abort the logic and keep returning `fault()` in commit c046df3. I want make each commit self-contained at least.

> 5. rename dispatch() to execute_open(). does this new name imply a state ?

Yes, I want to make it clear that this function belongs to the open state, which will be further refactored to "Formalize a connection state machine". Most likely, they will become the "do_execute()" member function of `XXXState` which inherites `Protocol::State`, like https://github.com/ceph/ceph/blob/5da58904d108cc81b7de12dbfb38d8e6d1c1a737/src/crimson/net/ProtocolV1.h#L92-L96 

> how are the states encapsulated? are they different continuations or classes or states after this change?

see 5.